tests/pytorch/fpga/vgg_conv_optimization_example.py

```diff
@@ -0,0 +1,421 @@
+#!/usr/bin/env python3
+
+# ----------------------------------------
+# VGG-16 Optimization Example
+# ----------------------------------------
+from dace.transformation.interstate import FPGATransformSDFG, InlineSDFG
+from daceml.transformation import InputToConstant
+
+import time
+import math
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import json
+import os
+
+import numpy as np
+
+import daceml.onnx as donnx
+import dace
+from daceml.pytorch import DaceModule, dace_module
+import copy
+
+from daceml.util import utils
+from dace.transformation.dataflow import streaming_memory as sm
+from dace.transformation.dataflow import PruneConnectors
+from dace.transformation.interstate import InlineSDFG
+from dace.sdfg import SDFG, SDFGState, trace_nested_access
+from dace import config, data as dt, dtypes, Memlet, symbolic
+from dace.sdfg import nodes as nd
+
+from dace.transformation.optimizer import Optimizer
+from dace.transformation.dataflow import streaming_memory as sm
+
+import argparse
+
+# Random numbers
+import random
+
+torch.manual_seed(0)
+random.seed(0)
+np.random.seed(0)
+
+# ----------------------------------------
+# Node Mappings
+# ----------------------------------------
+access_node_block_mapping = {
+    0: [27, 28, 29, 30],
+    1: [31, 32, 33, 34, 35],
+    2: [36, 37, 38, 39, 40, 41, 42],
+    3: [43, 44, 45, 46, 47, 48, 49],
+    4: [50, 51, 52, 53, 54, 55, 56],
+}
+
+conv_mapping = {
+    0: [0, 2],
+    1: [5, 7],
+    2: [10, 12, 14],
+    3: [17, 19, 21],
+    4: [24, 26, 28]
+}
+
+# ----------------------------------------
+# Operator configurations
+# ----------------------------------------
+net_config = {
+    "blocks": [{
+        "pe": 4,
+        "tile": 1120,
+        "vec": 16
+    }, {
+        "pe": 4,
+        "tile": 1120,
+        "vec": 16
+    }, {
+        "pe": 4,
+        "tile": 1120,
+        "vec": 8
+    }, {
+        "pe": 8,
+        "tile": 784,
+        "vec": 4
+    }, {
+        "pe": 8,
+        "tile": 196,
+        "vec": 2
+    }]
+}
+
+
+# ----------------------------------------
+# Memory bank RR-interleaving
+# ----------------------------------------
+def fpga_rr_interleave_containers_to_banks(sdfg: SDFG, num_banks: int = 4):
```

Comment:
Why are you not using the methods offered by dace?

Reply:
Using the DaCe implementation now